libkbfs/backpressure_disk_limiter.go

```diff
@@ -924,30 +997,31 @@ func (bdl *backpressureDiskLimiter) onBlocksFlush(
 	bdl.journalTracker.onBlocksFlush(blockBytes, chargedTo)
 }
 
-func (bdl *backpressureDiskLimiter) onBlocksDelete(
-	ctx context.Context, blockBytes, blockFiles int64) {
+func (bdl *backpressureDiskLimiter) releaseAndCommit(ctx context.Context,
+	typ diskLimitTrackerType, blockBytes, blockFiles int64) {
 	bdl.lock.Lock()
 	defer bdl.lock.Unlock()
-	bdl.journalTracker.onBlocksDelete(blockBytes, blockFiles)
-}
-
-func (bdl *backpressureDiskLimiter) onDiskBlockCacheDelete(
-	ctx context.Context, blockBytes int64) {
-	if blockBytes == 0 {
-		return
+	switch typ {
+	case journalLimitTracker:
+		bdl.journalTracker.releaseAndCommit(blockBytes, blockFiles)
+	case diskCacheLimitTracker:
+		bdl.diskCacheByteTracker.releaseAndCommit(blockBytes)
+	case syncCacheLimitTracker:
+		bdl.syncCacheByteTracker.releaseAndCommit(blockBytes)
 	}
-	bdl.lock.Lock()
-	defer bdl.lock.Unlock()
-	bdl.diskCacheByteTracker.onBlocksDelete(blockBytes)
+	bdl.overallByteTracker.releaseAndCommit(blockBytes)
 }
 
-func (bdl *backpressureDiskLimiter) beforeDiskBlockCachePut(
-	ctx context.Context, blockBytes int64) (
+func (bdl *backpressureDiskLimiter) reserve(
+	ctx context.Context, typ diskLimitTrackerType, blockBytes int64) (
 	availableBytes int64, err error) {
 	if blockBytes == 0 {
 		// Better to return an error than to panic in ForceAcquire.
-		return 0, errors.New("backpressureDiskLimiter.beforeDiskBlockCachePut" +
-			" called with 0 blockBytes")
+		return 0, errors.New("reserve called with 0 blockBytes")
+	}
+	tracker, err := bdl.trackerFromType(typ)
+	if err != nil {
+		return 0, errors.New("reserve called with invalid tracker type")
```

Comment:
Why construct a new error instead of using `err`?  Theoretically the error returned by `trackerFromType` could be anything, not just about an invalid tracker type.